spec.html

```diff
@@ -321,12 +326,17 @@ <h1>%TypedArray%.prototype.with ( _index_, _value_ )</h1>
                         1. Let _k_ be 0.
                         1. Repeat, while _k_ &lt; _len_,
                             1. Let _Pk_ be ! ToString(𝔽(_k_)).
-                            1. If _k_ is _actualIndex_, let _fromValue_ be _value_.
-                            1. Else, let _fromValue_ be ! Get(_O_, _Pk_).
-                            1. Perform ? Set(_A_, _Pk_, _fromValue_, *true*).
+                            1. If _k_ is _actualIndex_, then
+                                1. [id="step-typedarray-with-set"] Perform ? Set(_A_, _Pk_, _value_, *true*).
+                            1. Else,
+                                1. Let _fromValue_ be ! Get(_O_, _Pk_).
+                                1. Perform ! Set(_A_, _Pk_, _fromValue_, *true*).
                             1. Set _k_ to _k_ + 1.
                         1. Return _A_.
                     </emu-alg>
+                    <emu-note>
+                        Step <emu-xref href="#step-typedarray-with-set"></emu-xref> may return an abrupt completion because _value_ is a value of any ECMAScript language type and is passed to ToBigInt or ToNumber.
```

Comment:
can you provide some example input that would behave differently with this change?

Reply:
Not an observable change. Just an attempt to clarify that `Set` can throw for the one case where it uses the value that is being passed to the method, rather than existing values in the source array.

Similar to the added note to `toSpliced`, trying to clarify why only of the three calls to `Set` is marked with `?` rather than `!`.

This should behave the same before and after the change:

```js
(new Uint8Array(1)).with(0, 0); // OK
(new BigInt64Array(1)).with(0, 0n); // OK

(new Uint8Array(1)).with(0, 0n); // Uncaught TypeError: Cannot convert a BigInt value to a number
(new BigInt64Array(1)).with(0, 0); // Uncaught TypeError: Cannot convert 0 to a BigInt
```

Reply:
ah ok, gotcha. and, no change to the normal array "with", i assume

Reply:
yep, no change to normal array `with` as the value isn't introspected as part `CreateDataProperty`.